internal/services/recoveryservices/backup_protected_vm_resource.go

```diff
@@ -268,7 +308,18 @@ func resourceRecoveryServicesBackupProtectedVMWaitForDeletion(ctx context.Contex
 	}
 
 	// we should also wait for the operation to complete, or it will fail when creating a new backup vm with the same vm in different vault immediately.
-	opState := &pluginsdk.StateChangeConf{
+	opState := resourceRecoveryServicesBackupProtectedVMOperationRefreshFunc(ctx, opResultClient, d.Timeout(pluginsdk.TimeoutDelete), vaultName, resourceGroup, operationId)
```

Comment:
this timeout isn't going to be correct, you'll want to obtain the remaining time from the `ctx` and use:

```
deadline, ok := ctx.Deadline()
if !ok {
  return fmt.Errorf("internal-error: context had no deadline")
}
// you can then use `time.Until(deadline)`
```